### PR TITLE
Create endpoint in Agent Client Authorisation for the orchestrator to…

### PR DESCRIPTION
**Pull request by oarshad**
… call

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala [hunk 1]
```diff
@@ -72,6 +73,25 @@ class AgencyInvitationsController @Inject()(
     }
   }
 
+  def replaceUrnInvitationWithUtr(givenUrn: Urn, utr: Utr): Action[AnyContent] = Action.async { implicit request =>
+    invitationsService.findLatestInvitationByClientId(clientId = givenUrn.value) flatMap {
+      case Some(i) if i.status == Pending =>
+        invitationsService
+          .create(i.arn, i.clientType, i.service, utr, utr, i.origin)
```
Comment (geoffreywatson) [inline app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala:80]: i.service will be HMRC-TERSNT-ORG...we want HMRC-TERS-ORG for the new one correct?

File: app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala [hunk 2]
```diff
@@ -72,6 +73,25 @@ class AgencyInvitationsController @Inject()(
...
+          .flatMap(_ => invitationsService.cancelInvitation(i))
+          .map {
+            case Right(_)                          => Created
+            case Left(StatusUpdateFailure(_, msg)) => invalidInvitationStatus(msg)
+          }
+      case Some(i) if i.status == IAccepted && !i.isRelationshipEnded =>
+        invitationsService
+          .create(i.arn, i.clientType, i.service, utr, utr, i.origin)
```
Comment (geoffreywatson) [inline app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala:88]: same goes here; also we should call setRelationshipEnded on the existing Accepted (ended by "HMRC")

File: app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala (hunk 1 above)
Comment (oarshad) [reply app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala:80]: `HMRC-TERSNT-ORG` to `HMRC-TERS-ORG`
Yes, that is correct

File: app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala (hunk 2 above)
Comment (oarshad) [reply app/uk/gov/hmrc/agentclientauthorisation/controllers/AgencyInvitationsController.scala]: Fixed

**Pull request merged**